libfoundation/src/foundation-java-private.cpp

```diff
@@ -1464,6 +1733,61 @@ void MCJavaPrivateDestroyObject(MCJavaObjectRef p_object)
     
     s_env -> DeleteGlobalRef(t_obj);
 }
+
+void MCJavaPrivateDoNativeListenerCallback(jlong p_handler, jstring p_method_name, jobjectArray p_args)
+{
+    MCAutoStringRef t_method_name;
+    if (!__MCJavaStringFromJString(p_method_name, &t_method_name))
+        return;
+ 
+    MCValueRef t_handler = nullptr;
+    MCValueRef t_handlers = reinterpret_cast<MCValueRef>(p_handler);
+    if (MCValueGetTypeCode(t_handlers) == kMCValueTypeCodeArray)
+    {
+        // Array of handlers for interface proxy
+        MCNewAutoNameRef t_key;
+        if (!MCNameCreate(*t_method_name, &t_key) ||
+            !MCArrayFetchValue(static_cast<MCArrayRef>(t_handlers),
+                              false, *t_key, t_handler) ||
+            MCValueGetTypeCode(t_handler) != kMCValueTypeCodeHandler)
+        {
+            t_handler = nullptr;
+        }
+    }
+    else
+    {
+        MCAssert(MCValueGetTypeCode(t_handlers) == kMCValueTypeCodeHandler);
+        // Single handler for listener interface
+        t_handler = t_handlers;
+    }
+    
+    if (t_handler == nullptr)
+    {
+        MCErrorThrowGenericWithMessage(MCSTR("callback handler not found for listener method %{method}"),
```

Comment:
I'm a little ponderous over what happens to this error (if it occurs) - this JNI native method is called from Java, in many case without any LCB or LCS related stack-frames. So we are updating the error state without anything 'nearby' to handle it. Indeed, the same is true of an error thrown by what the handler calls.